### PR TITLE
rewrite 01171 test

### PR DESCRIPTION
**Pull request by CheSema**
01171_mv_select_insert_isolation_long.sh is flaky. Im trying to make it stable.
The main issue that it just works slowly on some heavy builds, like tsan/debug + S3 storage.
Closes: https://github.com/ClickHouse/clickhouse-private/issues/13090

Originally tests did 100 iterations in each thread. It takes a different time depends on build type.
Now it do some amount iterations for fixed time and check that at least `MIN_ITERATIONS` count iterations are done.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by CheSema** [comment]
Stateless tests flaky check (asan) --
It failed very good. I made comprehensive diagnostic.
```
2024-07-24 18:41:43 Reason: having stderror:  
2024-07-24 18:41:43 45/50 : not enough iterations of select_insert_action has been made from 1721838864 until 1721839294
2024-07-24 18:41:43 46/50 : not enough iterations of select_insert_action has been made from 1721838864 until 1721839294
```

Asan is a heavy build. The test worked 450 seconds and did 30 - 49 iterations. Which is less that threshold 50. I need just correct that threshold.

File: tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh [hunk 1]
```diff
@@ -110,36 +153,69 @@ function thread_select_insert()
...
+
+if [[ $((MAIN_TIME_PART + SECOND_TIME_PART + WAIT_FINISH + LAST_TIME_GAP)) -ge  600 ]]; then
+    echo "time sttings are wrong" 2>&1
+    exit 1
+fi
+
+START_TIME=$(get_now)
+STOP_TIME=$((START_TIME + MAIN_TIME_PART))
+SECOND_STOP_TIME=$((STOP_TIME + SECOND_TIME_PART))
+MIN_ITERATIONS=25
```
Comment (CheSema) [inline tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh:171]:
```suggestion
MIN_ITERATIONS=20
```

**Comment by CheSema** [comment]
Stateless Tests (msan) [2/4] ---
02481_async_insert_race_long consumed all the memory https://github.com/ClickHouse/clickhouse-private/issues/13101

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Ok. Btw, there is also the `$SECONDS` variable - the time since the script startup.

File: tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh [hunk 2]
```diff
@@ -110,36 +153,69 @@ function thread_select_insert()
...
-
-wait $PID_1 && wait $PID_2 && wait $PID_3
-kill -TERM $PID_4
-kill -TERM $PID_5
-kill -TERM $PID_6
-kill -TERM $PID_7
-kill -TERM $PID_8
-wait
-wait_for_queries_to_finish 40
+MAIN_TIME_PART=400
```
Comment (Algunenano) [inline tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh:158]: This has made the test 40x slower in release build (used to take 10 seconds, now it takes 400): https://play.clickhouse.com/play?user=play#U2VsZWN0ICogZnJvbSBjaGVja3Mgd2hlcmUgdGVzdF9uYW1lID0gJzAxMTcxX212X3NlbGVjdF9pbnNlcnRfaXNvbGF0aW9uX2xvbmcnIEFORCBjaGVja19uYW1lID0gJ1N0YXRlbGVzcyB0ZXN0cyAocmVsZWFzZSknIGFuZCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMCBvcmRlciBieSBjaGVja19zdGFydF90aW1lIGRlc2M=

And it's executed in sequential mode. So it's adding 6 minutes to all runs.

Comment (CheSema) [reply tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh:158]: Lest remove `no-parallel`. It should be fine.
I will do this.

Comment (CheSema) [reply tests/queries/0_stateless/01171_mv_select_insert_isolation_long.sh:158]: https://github.com/ClickHouse/ClickHouse/pull/67470

**Pull request merged**